### PR TITLE
docs(modal): card playground

### PR DESCRIPTION
**Pull request by sean-perkins**
Introduces the card style playground for `ion-modal`.

Decided to remove the full screen section (already cover heavily in the inline and controller examples) and the sizes section (changing modal height can be covered in the styling section).

### CODE REVIEW

File: docs/api/modal.md [hunk 1]
```diff
@@ -102,11 +96,13 @@ The `presentingElement` property accepts a reference to the element that should
...
-TODO: Playground Example
+:::note
+The card display style is only available on iOS.
+:::
 
-#### Custom sizes
+import CardExample from '@site/static/usage/modal/card/basic/index.md';
 
-TODO: Playground Example
+<CardExample />
```
Comment (amandaejohnston) [inline docs/api/modal.md:105]: On the JS Stackblitz, if I click the Close button (rather than swiping to close) I get an error (`modal is not defined`).

**Review by amandaejohnston — CHANGES_REQUESTED** [review]

**Review by amandaejohnston — APPROVED** [review]

File: static/usage/modal/card/basic/react.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        <IonButton id="open-modal" expand="block">
+          Open
+        </IonButton>
+        <IonModal
+          ref={modal}
+          trigger="open-modal"
+          swipeToClose={true}
+          presentingElement={document.querySelector('.ion-page')}
+          style={{
+            '--backdrop-opacity': '0.2',
```
Comment (liamdebeasi) [inline static/usage/modal/card/basic/react.md:42]: Why do we set the backdrop opacity?

**Review by liamdebeasi — COMMENTED** [review]
Also same comments as https://github.com/ionic-team/ionic-docs/pull/2323#pullrequestreview-966736895

**Review by liamdebeasi — APPROVED** [review]
Looks good. I saw a few other issues, but they do not appear to be caused by changes in this PR.

I created FW-1524 and FW-1525 to track these.

**Pull request merged**